assets/pages/oneoff-contributions/components/formFields.jsx

```diff
@@ -3,18 +3,111 @@
 // ----- Imports ----- //
 
 import React from 'react';
-import NameFormField from './nameFormField';
-import EmailFormFieldContainer from './emailFormFieldContainer';
+import { connect } from 'react-redux';
+import type { Dispatch } from 'redux';
+import TextInput from 'components/textInput/textInput';
+import {
+  setFullName,
+  setEmail,
+  type Action as UserAction,
+} from 'helpers/user/userActions';
+import { type UserFormFieldAttribute, formFieldIsValid, shouldShowError } from 'helpers/checkoutForm/checkoutForm';
+import {
+  type Action as CheckoutAction,
+  setFullNameShouldValidate,
+  setEmailShouldValidate,
+} from './contributionsCheckoutContainer/checkoutFormActions';
 
 
+// ----- Types ----- //
+
+type PropTypes = {
+  fullName: UserFormFieldAttribute,
+  email: UserFormFieldAttribute,
+  setFullName: (string) => void,
+  setEmail: (string) => void,
+  setFullNameShouldValidate: () => void,
+  setEmailShouldValidate: () => void,
+};
+
+// ----- Map State/Props ----- //
+
+function mapStateToProps(state) {
```

Comment:
Could you add a type annotation for the `state`? It should be the `PageState` from `oneOffContributionsReducer`.